mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m

```diff
@@ -0,0 +1,384 @@
+            
+            FIROptions *options = [[FIROptions alloc] initWithGoogleAppID:googleAppId GCMSenderID:gcmSenderId];
+            
+            self.firebaseOptions = options;
+            [FIRApp configureWithOptions:options];
+            
+            self->_started = YES;
+            
+            dispatch_async(dispatch_get_main_queue(), ^{
+                NSDictionary *userInfo = @{mParticleKitInstanceKey:[[self class] kitCode]};
+                
+                [[NSNotificationCenter defaultCenter] postNotificationName:mParticleKitDidBecomeActiveNotification
+                                                                    object:nil
+                                                                  userInfo:userInfo];
+            });
+        });
+    } else {
+        _started = YES;
+    }
+
+    return [self execStatus:MPKitReturnCodeSuccess];
+}
+
+- (id const)providerKitInstance {
+    return [self started] ? self : nil;
+}
+
+- (MPKitExecStatus *)logCommerceEvent:(MPCommerceEvent *)commerceEvent {
+    if ([FIRApp defaultApp] == nil) {
+        if (self.firebaseOptions != nil) {
+            [FIRApp configureWithOptions:self.firebaseOptions];
+        } else {
+            return [self execStatus:MPKitReturnCodeFail];
+        }
+    }
+    
+    NSDictionary<NSString *, id> *parameters = [[NSMutableDictionary alloc] init];
+    
+    switch (commerceEvent.action) {
+        case MPCommerceEventActionAddToCart: {
+            for (MPProduct *product in commerceEvent.products) {
+                parameters = [self getParameterForCommerceEvent:commerceEvent andProduct:product withValue:nil];
+                
+                [FIRAnalytics logEventWithName:kFIREventAddToCart
+                                    parameters:parameters];
+            }
+        }
+            break;
+            
+        case MPCommerceEventActionRemoveFromCart: {
+            for (MPProduct *product in commerceEvent.products) {
+                parameters = [self getParameterForCommerceEvent:commerceEvent andProduct:product withValue:nil];
+                
+                [FIRAnalytics logEventWithName:kFIREventRemoveFromCart
+                                    parameters:parameters];
+            }
+        }
+            break;
+            
+        case MPCommerceEventActionAddToWishList: {
+            for (MPProduct *product in commerceEvent.products) {
+                parameters = [self getParameterForCommerceEvent:commerceEvent andProduct:product withValue:nil];
+
+                [FIRAnalytics logEventWithName:kFIREventAddToWishlist
+                                    parameters:parameters];
+            }
+        }
+            break;
+            
+        case MPCommerceEventActionCheckout: {
+            NSNumber *value = @0;
+            for (MPProduct *product in commerceEvent.products) {
+                value = @([value doubleValue] + [product.price doubleValue]);
```

Comment:
Oooh good catch.

Reply:
Just in case Sam's comment below got lost in the shuffle, apparently this should use a diff API vs. manually multiplying by quantity: https://github.com/mparticle-integrations/mparticle-apple-integration-google-analytics-firebase/pull/1#discussion_r254314866

Reply:
yea. Anytime I was calculating for the whole transaction before, i am now using transactionAttributes.revenue